internal/network/none_platform_node_ips_allocation.go

```diff
@@ -0,0 +1,374 @@
+package network
+
+import (
+	"encoding/json"
+	stderrors "errors"
+	"net"
+	"sort"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	"golang.org/x/sys/unix"
+)
+
+// When none platform is in use, if there is ambiguity in node-ip assignment, then incorrect assignment might lead
+// to installation failure.  This happens when etcd detects that the socket address from an etcd node does not match
+// the expected address in the peer certificate. In this case etcd rejects such connection.
+// Example: assuming two networks - net1 and net2.
+// master node 1 has 1 address that belongs to net1.
+// master node 2 has 2 addresses.  one that belongs to net 1, and another that belongs to net 2
+// master node 3 has 1 address that belongs to net 1.
+// If the selected node-ip of master node 2 belongs to net 2, then when it will create a connection with any other master node,
+// the socket address will be the address that belongs to net 1. Since etcd expects it to be the same as the node-ip, it will
+// reject the connection.
+// A similar situation may happen with runnel IPs (Geneve for OVN).  When setting tunnel, OVS configuration contains the
+// node-ips as endpoints.  When there is a collision, the receiving endpoint of a tunnel may receive a tunnel packet
+// from an IP address it does expect.  This leads OVS to ignore the packet.  Such situation may occur for master and worker
+// nodes.  If such collision happens for workers, there is a possibility that the cluster will still be functional.
+//
+// This file attempts to solve the issue automatically.  Masters failures are considered non-recoverable, and therefore no
+// node-ip allocation will be done for them. For workers we try to find node-ips with the least number of conflicts as possible.
```

Comment:
You sure we need to handle workers ips at all? Not sure how workers are influenced at all, they don't reach etcd

Reply:
The solution is applicable for Geneve tunnel used for cluster networks.